src/server/service/diagnostics.rs

```diff
@@ -1337,26 +1342,29 @@ mod log {
                 log_iter.map(|m| m.get_time()).collect::<Vec<i64>>(),
                 expected
             );
-            let log_iter = LogIterator::new(
-                &log_file,
-                timestamp("2019/08/23 18:09:53.387 +08:00"),
-                std::i64::MAX,
-                1 << (LogLevel::Warn as usize),
-                vec![],
-            )
-            .unwrap();
-            let expected = vec![
-                "2019/08/23 18:09:58.387 +08:00",
-                "2019/08/23 18:09:59.387 +08:00",
-                "2019/08/23 18:10:06.387 +08:00",
-            ]
-            .iter()
-            .map(|s| timestamp(s))
-            .collect::<Vec<i64>>();
-            assert_eq!(
-                log_iter.map(|m| m.get_time()).collect::<Vec<i64>>(),
-                expected
-            );
+
+            for time in vec![0, std::i64::MAX].into_iter() {
+                let log_iter = LogIterator::new(
```

Comment:
Could we apply a builder pattern to `LogIterator`, seems the constructor of it has lots of parameters.

Reply:
Rest LGTM